sdk/src/main/java/com/bugsnag/android/Stacktrace.java

```diff
@@ -12,12 +15,14 @@
 
     private static final int STACKTRACE_TRIM_LENGTH = 200;
 
-    final Configuration config;
+    private final List<String> projectPackages = new ArrayList<>();
```

Comment:
The idea behind this is to create a copy of `String[] projectPackages` from the constructor parameter, so that the user can't modify the internal fields of our class. This can be simplified a little bit by assigning the return value of `Arrays.asList` to `projectPackages`.

Generally a [for-each loop](https://stackoverflow.com/a/25538709/5144991) is preferred in Java over an indexed for-loop, as it's a bit easier to read, and has benefits such as throwing a `ConcurrentModificationException` if the collection is mutated during iteration.

Behind the scenes, a for-each loop creates an [`Iterator`](https://developer.android.com/reference/java/util/Iterator) object, which is responsible for keeping track of how far the loop has progressed. Because object allocation has a cost, there can be a small performance benefit to using the indexed syntax in hot loops, but 99% of the time that's not a problem.